nep-12.mediawiki

```diff
@@ -0,0 +1,2098 @@
+
+While there may be several options for 3rd party wallet providers that will help them to facilitate these transactions, the is currently no common consensus on the consistency of their interfaces. With a lack of consistency in interfaces to use these wallet provider services, dApp developers will be forced to make a decision to have their platform supported by a single provider, or to double or even triple their development efforts to accommodate all the different wallet provider interfaces. This will lead to a fragmentation in the dApps ecosystem.
+
+===Wallet providers===
+
+Each wallet provider, when deciding on supporting dApps to utilize their services as an authentication mechanism will be faced with a decision on how to implement an API to communicate with the dApps. Wallet providers can choose to create their own API from scratch, create their own version of existing projects, or aim to directly duplication an existing API. In the case that the provider decides to make their own API interface from scratch, and try to promote dApps to use it, time and effort will inevitably be wasted by both the provider and competing providers on getting dApp developers on board with using their custom communication interface. If we have a unified interface for such transactions, providers can spend more time on making their individual services better for their users.
+
+The current list of wallet providers that can benefit from the use of this protocol are currently:
+* NEL Chrome extension
+* nOS dApps browser
+* NeoLink Chrome extension
+* NEX Chrome extension
+* O3 dApps browser
+
+Each wallet provider has their own value proposition to it's users beyond the interface from this protocol itself, so it seems that formalizing it would be a net positive for all.
+
+Additionally, since there are a significant amount of overlap in the protocols between NEO and Ontology for sending assets, and interacting with contract on NeoVM, this proposal is a joint effort with [https://github.com/ontio/OEPs/pull/8 OEP6]. Since there are differences in the two platforms, the result of these proposals will not be identical, nor should they be, but getting as much overlap as possible will help to simplify cross-chain interactions for both platforms.
+
+
+==Specification==
+
+'''Provider Request Handling'''
+
+Implementation details of per dApp domain handling will be left to the discretion of the wallet provider. Providers can choose to allow users to trust certain dApps to automatically execute transaction on their behalf, or a certain subset of transaction types. Below are basic guidelines for general handling of requests from dApps via the protocol interface.
```

Comment:
That is a great paragraph.

It gives the notion that the provider can also automatically sign transactions if it marks a dApp as trusted, right?
However, for hardware wallets that would be a little bit more trick, because we need physical confirmation, right?

Reply:
That is correct. The proposal attempts to standardize on interface of the methods and general input and output behaviors. However doesn't not look to set standards on the approach for prompting the user for their input. So if the wallet has access to the users private key, and has formed an understanding with the user on how auto signing of certain dapps will work, then they are free to do so. But yes, for hardware wallets, we will still require the user approval for every transaction.